hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/AvroSchemaEvolutionUtils.java

```diff
@@ -113,6 +120,21 @@ public static InternalSchema reconcileSchema(Schema incomingSchema, InternalSche
       typeChange.updateColumnType(col, inComingInternalSchema.findType(col));
     });
 
+    // mark columns missing from incoming schema as nullable
+    Set<String> visited = new HashSet<>();
```

Comment:
hey @jonvex : when we added support for "hoodie.write.set.null.for.missing.columns", how did we miss this. the java docs for this method does call out that nulls will be injected for missing cols. but prior to this patch, I don't see that happening.

Reply:
do you think we can rename 2 variable names to make it more comprehensible or w/o ambiguity. 

diffFromOldSchema -> missingFromIncoming
diffFromEvolutionColumns -> additionsToTableSchema or additionsToOldSchema

Reply:
I like using `incoming` and `tableSchema` personally, I can change this

Reply:
@nsivabalan the current behavior will set null values for the missing data in the rows themselves but it does not set the column as nullable in the schema, that's what this patch is aiming to fix.

Reply:
gotcha.